pkg/apis/gcp/validation/worker.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp"
+	"github.com/gardener/gardener/pkg/apis/core"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var validVolumeLocalSSDInterfacesTypes = sets.NewString("NVME", "SCSI")
+
+// ValidateWorkerConfig validates a WorkerConfig object.
+func ValidateWorkerConfig(workers []core.Worker, decoder runtime.Decoder, decodeWorkerConfig func(decoder runtime.Decoder, worker *core.ProviderConfig, fldPath *field.Path) (*gcp.WorkerConfig, error), fldPath *field.Path) field.ErrorList {
```

Comment:
Previously, i did it by the way you said. But i found there is a questions to test.
Because if the code follow the style you said, the `ValidateWorkerConfig ` just contains the logic of `WorkerConfig`, it can not test the `DataVolume`.

So i think this style can cover more cases `DataVolume` and `WorkerConfig`.

Reply:
Accept